src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -662,19 +662,50 @@ public function setAutowiringTypes(array $types)
      */
     public function isAutowired()
     {
-        return $this->autowired;
+        return !empty($this->autowiredMethods);
+    }
+
+    /**
+     * Gets autowired methods.
+     *
+     * @return string[]
+     */
+    public function getAutowiredMethods()
+    {
+        return $this->autowiredMethods;
     }
 
     /**
      * Sets autowired.
      *
+     * Allowed values:
+     *   - true: constructor autowiring, same as $this->setAutowiredMethods(array('__construct'))
+     *   - false: no autowiring, same as $this->setAutowiredMethods(array())
+     *
      * @param bool $autowired
      *
      * @return Definition The current instance
      */
     public function setAutowired($autowired)
```

Comment:
I propose to deprecate it in another PR (if it's worth it) because we need determine how we deal with symfony/symfony#20648